### PR TITLE
Expose types of GraphNode to Lua

### PR DESCRIPTION
**Pull request by kermeow**
back again with another lua related pull request

- Exposes graph_node.h
- `cast_graph_node(node)` casts a generic GraphNode to its corresponding type

theoretically adds a lot more possibilities with gfx related lua modding if modders figure out how to traverse the graph properly

### CODE REVIEW

**Review by AgentXLP — APPROVED** [review]
Looks good

File: src/pc/lua/smlua_functions.c [hunk 1]
```diff
@@ -946,6 +946,69 @@ int smlua_func_collision_find_surface_on_ray(lua_State* L) {
     return 1;
 }
 
+  ////////////////
+ // graph node //
+////////////////
+
+typedef struct { s16 type; u16 lot; } GraphNodeLot;
+static GraphNodeLot graphNodeLots[23] = {
+    { GRAPH_NODE_TYPE_ANIMATED_PART, LOT_GRAPHNODEANIMATEDPART },
```
Comment (Isaac0-dev) [inline src/pc/lua/smlua_functions.c:955]: these lot constants are autogenerated and shouldn't be used directly in C. those constants should be made non autogenerated

File: src/pc/lua/smlua_functions.c [hunk 2]
```diff
@@ -946,6 +946,69 @@ int smlua_func_collision_find_surface_on_ray(lua_State* L) {
     return 1;
 }
 
+  ////////////////
+ // graph node //
+////////////////
+
+typedef struct { s16 type; u16 lot; } GraphNodeLot;
+static GraphNodeLot graphNodeLots[23] = {
```
Comment (Isaac0-dev) [inline src/pc/lua/smlua_functions.c:954]: I recommend removing the manual count of graph node lots here, the compiler can automatically count this for you, and that would be useful for letting people add stuff to this list without accidentally forgetting to change this value

File: src/pc/lua/smlua_functions.c [hunk 3]
```diff
@@ -946,6 +946,69 @@ int smlua_func_collision_find_surface_on_ray(lua_State* L) {
...
+    { GRAPH_NODE_TYPE_TRANSLATION_ROTATION, LOT_GRAPHNODETRANSLATIONROTATION },
+};
+
+int smlua_func_cast_graph_node(lua_State* L) {
+    if (!smlua_functions_valid_param_count(L, 1)) { return 0; }
+
+    struct GraphNode* graphNode;
+
+    if (smlua_is_cobject(L, 1, LOT_GRAPHNODE)) {
+        graphNode = (struct GraphNode*)smlua_to_cobject(L, 1, LOT_GRAPHNODE);
```
Comment (Isaac0-dev) [inline src/pc/lua/smlua_functions.c:986]: `LOT_GRAPHNODE` is autogenerated

File: src/pc/lua/smlua_functions.c [hunk 4]
```diff
@@ -946,6 +946,69 @@ int smlua_func_collision_find_surface_on_ray(lua_State* L) {
...
+        if (!gSmLuaConvertSuccess) { LOG_LUA("cast_graph_node: Failed to convert parameter 1"); return 0; }
+    } else if (smlua_is_cobject(L, 1, LOT_FNGRAPHNODE)) {
+        graphNode = (struct GraphNode*)smlua_to_cobject(L, 1, LOT_FNGRAPHNODE);
```
Comment (Isaac0-dev) [inline src/pc/lua/smlua_functions.c:989]: `LOT_FNGRAPHNODE` is autogenerated

**Review by Isaac0-dev — COMMENTED** [review]
sorry for the late review but can these issues please be fixed?
I dont have time to test your pr but here's some things i noticed

**Pull request merged**